src/NUnitFramework/framework/Internal/RuntimeFramework.cs

```diff
@@ -98,6 +107,23 @@ public sealed class RuntimeFramework
                         break;
                 }
             }
+            else if (isNetCore)
+            {
+#if NETSTANDARD2_0
+                string frameworkName = Assembly.GetEntryAssembly()?.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName;
```

Comment:
This tells you the version of .NET Core that the entry assembly was compiled against (e.g. 1.0) but not the version it's running on (could be any other version).

Reply:
That's true... I didn't realize that. I'm not sure of a way to get the running .NET Core version then. I'll investigate that issue further.

Reply:
So looking into how other projects do it, BenchmarkDotNet had the following issue: https://github.com/dotnet/BenchmarkDotNet/issues/448 - however, it doesn't work for docker or self-contained distributions (which is a big issue for me - all of my .net core projects use self-contained) - see the open https://github.com/dotnet/BenchmarkDotNet/issues/788. So I'm not sure there is a really good way to detect it at runtime.

Reply:
I see. Can we detect *whether* it's .NET Core while sidestepping the whole version issue, just so we don't restrict our future options? So `netcore` would work, but `netcore2.1` would not be recognized?

Reply:
We can do that, yeah. Is that the route we want to go (limit to only netcore - no version detection) or do we want to implement the BenchmarkDotNet way that will at least detect the version in some cases like they do?

Reply:
I'm not sure, if the check would fail if the tests are compiled to a self-contained app or run in Docker. @nunit/framework-team?